mass_mailing_list_dynamic/tests/test_dynamic_lists.py

```diff
@@ -82,9 +85,22 @@ def test_sync_when_sending_mail(self):
             "category_id": [(4, self.tag.id, False)],
             "email": "extra@example.com",
         })
-        # Before sending the mail, the list is updated
-        with patch("odoo.addons.base.ir.ir_mail_server"
-                   ".IrMailServer.send_email") as send_email:
+        # Mock sending low level method, because an auto-commit happens there
+        with patch("odoo.addons.mail.models.mail_mail.MailMail.send"):
             self.mail.send_mail()
-            self.assertEqual(6, send_email.call_count)
```

Comment:
It doesn't make sense, as I have changed the mocked call (because yours was incorrect as was causing a commit to be performed). Any way, I let the assert with the count=1, that is the one needed.